packages/arb-token-bridge-ui/src/hooks/useArbTokenBridge.ts

```diff
@@ -164,60 +169,89 @@ export const useArbTokenBridge = (
   const depositEth = async ({
     amount,
     l1Signer,
-    txLifecycle
+    txLifecycle,
+    destinationAddress
   }: {
     amount: BigNumber
     l1Signer: Signer
-    txLifecycle?: L1EthDepositTransactionLifecycle
+    txLifecycle?:
+      | L1EthDepositTransactionLifecycle & L1ContractCallTransactionLifecycle
+    destinationAddress?: string
   }) => {
     const ethBridger = await EthBridger.fromProvider(l2.provider)
 
-    let tx: L1EthDepositTransaction
+    let tx: L1EthDepositTransaction & L1ContractCallTransaction
+    let receipt:
+      | L1EthDepositTransactionReceipt
+      | L1ContractCallTransactionReceipt
+
+    const senderAddress = await l1Signer.getAddress()
 
     try {
-      tx = await ethBridger.deposit({
-        amount,
-        l1Signer
-      })
+      tx = (
+        destinationAddress && destinationAddress !== senderAddress
+          ? // send ETH to a custom address
+            await ethBridger.depositTo({
+              amount,
+              l1Signer,
+              l2Provider: l2.provider,
+              destinationAddress
+            })
+          : await ethBridger.deposit({
+              amount,
+              l1Signer
+            })
+      ) as typeof tx
 
       if (txLifecycle?.onTxSubmit) {
         txLifecycle.onTxSubmit(tx)
       }
+      addTransaction({
```

Comment:
Any reason why we moved the `addTransaction` and `receipt` call a few lines above?
If it's an optimisation, I'd keep it for a separate PR and keep the diffs minimum in this one since it's critical.